### PR TITLE
Feature: Migration to WinAppSDK

### PR DESCRIPTION
**Pull request by yaichenbaum**
**Validation**
How did you test these changes?
- Pipeline does not work but app builds

Before merging, we should review every file and take note of whatever needs to be added back.
Progress is being tracked here, please report any issues: #9780

### CODE REVIEW

File: src/Files.Uwp/App.xaml [hunk 1]
```diff
@@ -15,8 +15,12 @@
             <!--  Default nav menu item height  -->
             <x:Double x:Key="NavigationViewItemOnLeftMinHeight">32</x:Double>
 
+            <!--  Fix caption buttons background  -->
+            <SolidColorBrush x:Key="WindowCaptionBackground" Color="Transparent" />
+            <SolidColorBrush x:Key="WindowCaptionBackgroundDisabled" Color="Transparent" />
+
             <ResourceDictionary.MergedDictionaries>
-                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" ControlsResourcesVersion="Version2" />
+                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
```
Comment (yaichenbaum) [inline src/Files.Uwp/App.xaml:23]: Do we still need this reference with WinUI 3?

File: src/Files.Uwp/App.xaml.cs [hunk 2]
```diff
@@ -230,362 +231,72 @@ static async Task OptionalTask(Task task, bool condition)
...
-            _ = InitializeAppComponentsAsync().ContinueWith(t => Logger.Warn(t.Exception, "Error during InitializeAppComponentsAsync()"), TaskContinuationOptions.OnlyOnFaulted);
-
-            var rootFrame = EnsureWindowIsInitialized();
-
-            if (e.PrelaunchActivated == false)
+            if (activatedEventArgs.Data is Windows.ApplicationModel.Activation.IActivatedEventArgs iaea)
             {
-                if (canEnablePrelaunch)
-                {
-                    TryEnablePrelaunch();
```
Comment (yaichenbaum) [inline src/Files.Uwp/App.xaml.cs:262]: I hear that Microsoft is planning to add pre-launch apis to WinAppSdk, something to keep in mind here in case we want to comment out the code instead of removing it.

File: src/Files.Uwp/App.xaml.cs [hunk 3]
```diff
@@ -209,17 +210,17 @@ private static async Task InitializeAppComponentsAsync()
...
                 userSettingsService.ReportToAppCenter();
             });
 
             // Check for required updates
-            var updateService = Ioc.Default.GetRequiredService<IUpdateService>();
-            await updateService.CheckForUpdates();
-            await updateService.DownloadMandatoryUpdates();
+            //var updateService = Ioc.Default.GetRequiredService<IUpdateService>();
+            //await updateService.CheckForUpdates(); // WINUI3
+            //await updateService.DownloadMandatoryUpdates();
```
Comment (yaichenbaum) [inline src/Files.Uwp/App.xaml.cs:223]: Can we re-enable this update service?

File: src/Files.Uwp/App.xaml.cs [hunk 4]
```diff
@@ -230,362 +231,72 @@ static async Task OptionalTask(Task task, bool condition)
...
         protected override async void OnLaunched(LaunchActivatedEventArgs e)
         {
+            var activatedEventArgs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
+
             await logWriter.InitializeAsync("debug.log");
-            Logger.Info($"App launched. Prelaunch: {e.PrelaunchActivated}");
+            Logger.Info($"App launched. Launch args type: {activatedEventArgs.Data.GetType().Name}");
 
             //start tracking app usage
-            SystemInformation.Instance.TrackAppUse(e);
```
Comment (yaichenbaum) [inline src/Files.Uwp/App.xaml.cs:249]: Was this moved somewhere else?

File: src/Files.Uwp/App.xaml.cs (hunk 2 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs:262]: Interesting, though I'm not sure if it ever worked for Files. Code still available in Apo.xaml.cs.old.

File: src/Files.Uwp/App.xaml.cs.old [hunk 5]
```diff
@@ -0,0 +1,758 @@
+using CommunityToolkit.Mvvm.DependencyInjection;
```
Comment (yaichenbaum) [inline src/Files.Uwp/App.xaml.cs.old:1]: What's this file for?

File: src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs [hunk 6]
```diff
@@ -20,6 +21,18 @@ public DynamicDialogResult DynamicResult
             get => ViewModel.DynamicResult;
         }
 
+        public new async Task<ContentDialogResult> ShowAsync() => await SetContentDialogRoot(this).ShowAsync();
+
+        // WINUI3
```
Comment (yaichenbaum) [inline src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs:26]: What does this comment indicate?

File: src/Files.Uwp/Helpers/AppServiceConnectionHelper.cs [hunk 7]
```diff
@@ -25,33 +25,7 @@ public static class AppServiceConnectionHelper
...
-        private async static void OnSuspending(object sender, SuspendingEventArgs e)
-        {
-            var deferral = e.SuspendingOperation.GetDeferral();
-            var nullConn = Task.FromResult<NamedPipeAsAppServiceConnection>(null);
-            ConnectionChanged?.Invoke(null, nullConn);
-            (await Instance)?.SendMessageAsync(new ValueSet() { { "Arguments", "Terminate" } });
-            (await Instance)?.Dispose();
-            Instance = nullConn;
-            deferral.Complete();
+            // WINUI3: app does not get suspended
```
Comment (yaichenbaum) [inline src/Files.Uwp/Helpers/AppServiceConnectionHelper.cs:28]: Can we remove this comment?

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs:223]: It crashes the app soon after start, to be checked why.

File: src/Files.Uwp/App.xaml.cs (hunk 2 above)
Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml.cs:262]: We've always had issues with it, but we know the code was good since we could manually pre-launch the app from visual studio. Either way, we can always copy this back in from an older commit.

File: src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs (hunk 6 above)
Comment (gave92) [reply src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs:26]: Nothing much, it only marks that the following method was added automatically when running the converter from uwp to winui3.

File: src/Files.Uwp/App.xaml.cs.old (hunk 5 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs.old:1]: Can be removed. I think we ported everything from it except prelaunch.

File: src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs (hunk 6 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Dialogs/DynamicDialog.xaml.cs:26]: Interesting, guess it does that to let you know it was behind the change.

File: src/Files.Uwp/App.xaml.cs.old (hunk 5 above)
Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml.cs.old:1]: I noticed that continue where you left off is broken, is that part of the launch arguments?

File: src/Files.Uwp/App.xaml (hunk 1 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml:23]: It appears in many win app sdk examples. Haven't tried to remove it.

File: src/Files.Uwp/App.xaml.cs.old (hunk 5 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs.old:1]: No, open tabs are saved to LocalSettings in App.cs/SaveSessionTabs() and restores in MainPageViewModel/OnNavigatedTo() (or they are supposed to). The code for it is still in place.

File: src/Files.Uwp/App.xaml (hunk 1 above)
Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml:23]: Let's keep it as is 👍

File: src/Files.Uwp/Helpers/FileExtensionHelpers.cs [hunk 8]
```diff
@@ -66,7 +66,7 @@ public static bool IsBrowsableZipFile(string filePath, out string ext)
             }
 
             ext = new[] { ".zip", ".7z", ".rar", ".tar" } // Only ext we want to browse
-                .FirstOrDefault(x => filePath.Contains(x, StringComparison.OrdinalIgnoreCase));
+                .FirstOrDefault(x => filePath?.Contains(x, StringComparison.OrdinalIgnoreCase) ?? false);
```
Comment (yaichenbaum) [inline src/Files.Uwp/Helpers/FileExtensionHelpers.cs:69]: Good catch

File: src/Files.Uwp/Helpers/FilePropertiesHelpers.cs [hunk 9]
```diff
@@ -60,65 +56,63 @@ public static async Task OpenPropertiesWindowAsync(object item, IShellPage assoc
...
+                    // Set min size
+                    propertiesWindow.MinWidth = 460;
+                    propertiesWindow.MinHeight = 550;
+
+                    // Set backdrop
+                    propertiesWindow.Backdrop = new WinUIEx.MicaSystemBackdrop() { DarkTintOpacity = 0.8 };
 
-                    bool windowShown = await appWindow.TryShowAsync();
-                    if (windowShown)
+                    if (AppWindowTitleBar.IsCustomizationSupported())
```
Comment (yaichenbaum) [inline src/Files.Uwp/Helpers/FilePropertiesHelpers.cs:83]: @ahmed605 it looks like we lost your work here, any ideas on why this isn't working to hide the minimize/maximize buttons? Not sure if this is related but the properties window no longer opens up next to the mouse cursor.

File: src/Files.Uwp/Interacts/BaseLayoutCommandImplementationModel.cs [hunk 10]
```diff
@@ -365,10 +365,9 @@ public virtual void ShareItem(RoutedEventArgs e)
         {
             DataTransferManager manager = DataTransferManager.GetForCurrentView();
             manager.DataRequested += new TypedEventHandler<DataTransferManager, DataRequestedEventArgs>(Manager_DataRequested);
-            DataTransferManager.ShowShareUI(new ShareUIOptions
-            {
-                Theme = Enum.IsDefined(typeof(ShareUITheme), ThemeHelper.RootTheme.ToString()) ? (ShareUITheme)ThemeHelper.RootTheme : ShareUITheme.Default
-            });
+
+                Windows.ApplicationModel.DataTransfer.DataTransferManager.As<UWPToWinAppSDKUpgradeHelpers.IDataTransferManagerInterop>().ShowShareUIForWindow(App.WindowHandle)
```
Comment (yaichenbaum) [inline src/Files.Uwp/Interacts/BaseLayoutCommandImplementationModel.cs:369]: Does this support passing the theme in?

File: src/Files.Uwp/Themes/Discord.xaml [hunk 11]
```diff
@@ -1,53 +0,0 @@
-
-<ResourceDictionary
```
Comment (yaichenbaum) [inline src/Files.Uwp/Themes/Discord.xaml:2]: Why was this removed?

File: src/Files.Uwp/UserControls/AddressToolbar.xaml [hunk 12]
```diff
@@ -519,53 +518,54 @@
...
+                                    MinHeight="300"
+                                    MaxHeight="400"
+                                    x:FieldModifier="public"
+                                    IsTabStop="True"
+                                    OngoingTasksViewModel="{x:Bind OngoingTasksViewModel}" />
+                            </Flyout>
+                        </Button.Flyout>
+                    </Button>
+                    <!--  WINUI3  -->
+                    <!--<InfoBadge
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/AddressToolbar.xaml:563]: Is info badge not supported?

File: src/Files.Uwp/Helpers/FilePropertiesHelpers.cs (hunk 9 above)
Comment (gave92) [reply src/Files.Uwp/Helpers/FilePropertiesHelpers.cs:83]: Hiding buttons apparently not supported: 
>These UWP AppWindow APIs have no direct 1:1 mapping to a Windows App SDK API.
[AppWindowTitleBar.SetPreferredVisibility](https://docs.microsoft.com/en-us/uwp/api/windows.ui.windowmanagement.appwindowtitlebar.setpreferredvisibility). 

https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/windowing

> Properties window no longer opens up next to the mouse cursor

Code commented out, need to be ported (I was hoping winui3 did it by default)

File: src/Files.Uwp/UserControls/FilePreviews/HtmlPreview.xaml [hunk 13]
```diff
@@ -14,7 +14,7 @@
         HorizontalAlignment="Stretch"
         VerticalAlignment="Stretch"
         CornerRadius="{StaticResource ControlCornerRadius}">
-        <WebView
+        <WebView2
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/FilePreviews/HtmlPreview.xaml:17]: Are we bundling WebView2 with the app or relying on the user installing it?

File: src/Files.Uwp/Themes/Discord.xaml (hunk 11 above)
Comment (gave92) [reply src/Files.Uwp/Themes/Discord.xaml:2]: All custom themes removed, some were giving compile errors (maybe related to mica/acrylic)

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml [hunk 14]
```diff
@@ -592,10 +591,23 @@
...
+            <TabView.TabStripFooter>
+                <Grid>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="Auto" />
+                        <ColumnDefinition Width="*" />
+                    </Grid.ColumnDefinitions>
+                    <ContentPresenter Content="{x:Bind ActionsControl, Mode=OneWay}" />
+
+                    <!--  This defines the drag area for the title bar  -->
+                    <Grid
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:604]: I think we need to keep the drag area on the home page, otherwise it overlaps with the tabs.

File: src/Files.Uwp/Helpers/FileExtensionHelpers.cs (hunk 8 above)
Comment (gave92) [reply src/Files.Uwp/Helpers/FileExtensionHelpers.cs:69]: We ended up fixing this issue twice, on main and here. No need for ` filePath?` since above we check  `if (string.IsNullOrEmpty(filePath))`.

File: src/Files.Uwp/UserControls/SidebarControl.xaml [hunk 15]
```diff
@@ -65,17 +64,18 @@
...
-                    </muxc:NavigationViewItem.Icon>
-                    <muxc:NavigationViewItem.InfoBadge>
-                        <muxc:InfoBadge Style="{ThemeResource CautionIconInfoBadgeStyle}" Visibility="{x:Bind IsInvalid, Mode=OneWay}" />
-                    </muxc:NavigationViewItem.InfoBadge>
-                </muxc:NavigationViewItem>
+                    <NavigationViewItem.Icon>
+                        <ImageIcon Source="{x:Bind Icon, Mode=OneWay}" />
+                    </NavigationViewItem.Icon>
+                    <!--  WINUI3  -->
+                    <!--<NavigationViewItem.InfoBadge>
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/SidebarControl.xaml:71]: Same question regarding info badge, we should add these to the WinUI progress issue.

File: src/Files.Uwp/UserControls/SidebarControl.xaml.cs [hunk 16]
```diff
@@ -1072,7 +1073,7 @@ private void Border_PointerExited(object sender, PointerRoutedEventArgs e)
         {
             if (!dragging) // keep showing pressed event if currently resizing the sidebar
             {
-                Window.Current.CoreWindow.PointerCursor = new CoreCursor(CoreCursorType.Arrow, 0);
+                //App.Window.CoreWindow.PointerCursor = new CoreCursor(CoreCursorType.Arrow, 0); //WINUI3
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: What's happening here?

File: src/Files.Uwp/Helpers/FilePropertiesHelpers.cs (hunk 9 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Helpers/FilePropertiesHelpers.cs:83]: We should remove the code for that, we currently have two close buttons on the properties window.

File: src/Files.Uwp/Themes/Discord.xaml (hunk 11 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Themes/Discord.xaml:2]: I'd like to keep them in the repo, at the very least we can exlude them from the project for now.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs [hunk 17]
```diff
@@ -93,26 +97,29 @@ private void Properties_Unloaded(object sender, RoutedEventArgs e)
...
                 RequestedTheme = selectedTheme;
                 if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 8))
                 {
                     switch (RequestedTheme)
                     {
                         case ElementTheme.Default:
-                            TitleBar.ButtonHoverBackgroundColor = (Color)Application.Current.Resources["SystemBaseLowColor"];
-                            TitleBar.ButtonForegroundColor = (Color)Application.Current.Resources["SystemBaseHighColor"];
+                            //WINUI3
+                            //TitleBar.ButtonHoverBackgroundColor = (Color)Application.Current.Resources["SystemBaseLowColor"];
```
Comment (yaichenbaum) [inline src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: No longer supported?

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml [hunk 18]
```diff
@@ -35,14 +35,6 @@
                 x:Key="BoolToWidthConverter"
                 FalseValue="0"
                 TrueValue="*" />
-
-            <AcrylicBrush
```
Comment (yaichenbaum) [inline src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml:39]: What's going on here?

File: src/Files.Uwp/UserControls/SidebarControl.xaml.cs (hunk 16 above)
Comment (gave92) [reply src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: Can't use CoreWindow on WinUi3. Need to figure out replacement.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs (hunk 17 above)
Comment (d2dyno1) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: Supported with AppWindow.TitleBar - however it only works on w11

Comment (gave92) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: Actually it is: `appWindow.TitleBar.ButtonHoverBackgroundColor` appears to work. Can be uncommented.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml (hunk 18 above)
Comment (gave92) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml:39]: Acrylic brush defined but not used anywhere, so removed.

File: src/Files.Uwp/Views/ColumnShellPage.xaml.cs [hunk 19]
```diff
@@ -172,7 +177,14 @@ public ColumnShellPage()
...
             ToolbarViewModel.PathBoxItemDropped += ColumnShellPage_PathBoxItemDropped;
-            ToolbarViewModel.BackRequested += ColumnShellPage_BackNavRequested;
+
+            /*
+              
+            TODO UA307 Default back button in the title bar does not exist in WinUI3 apps.
+            The tool has generated a custom back button in the MainWindow.xaml.cs file.
+            Feel free to edit its position, behavior and use the custom back button instead.
+            Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/case-study-1#restoring-back-button-functionality
+            */ToolbarViewModel.BackRequested += ColumnShellPage_BackNavRequested;
```
Comment (yaichenbaum) [inline src/Files.Uwp/Views/ColumnShellPage.xaml.cs:187]: Wasn't this our own implementation, I think we can revert this.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs (hunk 17 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: We're using experimental so I think it should work

File: src/Files.Uwp/UserControls/SidebarControl.xaml.cs (hunk 16 above)
Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: What does this break? Can you add it to the progress issue?

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml (hunk 14 above)
Comment (gave92) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:604]: What do you mean "it overlaps with the tabs"? I moved it out of main page because otherwise the tabs and "+" button were not clickable. Putting the drag area element behind (z-order) other elements may not work:
> The rectangular area occupied by the specified element acts as the title bar for pointer purposes, even if the element is blocked by another element, or the element is transparent.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs (hunk 17 above)
Comment (gave92) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: I'll check but I think it still crashed on w10

File: src/Files.Uwp/UserControls/SidebarControl.xaml.cs (hunk 16 above)
Comment (gave92) [reply src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: I think it causes this (already in the issue):
- [ ]  Grid splitters have the wrong cursor

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml (hunk 14 above)
Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:604]: UWP had a weird setup with this and I'm not sure how much changed but I'll take a look. As it is right now, tabs aren't clickable (there is a small section at the bottom of the title bar that works.

File: src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs (hunk 17 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Views/Pages/PropertiesSecurityAdvanced.xaml.cs:109]: Fair enough, they're probably still working on backporting it but I think it's planned for the next stable release.

File: src/Files.Uwp/UserControls/SidebarControl.xaml.cs (hunk 16 above)
Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: Thanks!

Comment (gave92) [reply src/Files.Uwp/UserControls/SidebarControl.xaml.cs:1076]: Replacement may look somethink like: `this.ProtectedCursor = Microsoft.UI.Input.InputSystemCursor.Create(Microsoft.UI.Input.InputSystemCursorShape.SizeWestEast)`

File: src/Files.Uwp/UserControls/AddressToolbar.xaml (hunk 12 above)
Comment (gave92) [reply src/Files.Uwp/UserControls/AddressToolbar.xaml:563]: Uncommenting causes crash :/

File: src/Files.Uwp/ServicesImplementation/DialogService.cs [hunk 20]
```diff
@@ -30,37 +32,36 @@ public DialogService()
...
             where TViewModel : class, INotifyPropertyChanged
         {
             try
             {
                 return await GetDialog(viewModel).ShowAsync();
             }
-            catch (Exception)
+            catch (Exception ex)
             {
+                _ = ex;
```
Comment (yaichenbaum) [inline src/Files.Uwp/ServicesImplementation/DialogService.cs:64]: Can we log this exception?

File: src/Files.Uwp/UserControls/AddressToolbar.xaml (hunk 12 above)
Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/AddressToolbar.xaml:563]: Not a blocker but I added to the issue

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml.cs:223]: I think this is a blocker :thinking:

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml (hunk 14 above)
Comment (gave92) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:604]: Funny I can click on the tabs just fine.

Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:604]: I'll check again but perhaps Win10 is different

File: src/Files.Uwp/Views/ColumnShellPage.xaml.cs (hunk 19 above)
Comment (gave92) [reply src/Files.Uwp/Views/ColumnShellPage.xaml.cs:187]: Lol the converter added this automatically. The problem is were this was commented (support for system back button): `SystemNavigationManager.GetForCurrentView().BackRequested`

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (puppetsw) [reply src/Files.Uwp/App.xaml.cs]: I can have a look at this and see if I can figure out what's going on.

**Comment by hez2010** [comment]
I've bumped the dotnet sdk version to adopt some bug/security fixes.

File: src/BackgroundTasks/BackgroundTasks.csproj [hunk 21]
```diff
@@ -1,140 +1,80 @@
...
-    <ProjectTypeGuids>{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <AllowCrossPlatformRetargeting>false</AllowCrossPlatformRetargeting>
   </PropertyGroup>
+  <PropertyGroup>
+    <CsWinRTComponent>true</CsWinRTComponent>
+    <CsWinRTWindowsMetadata>10.0.22000.0</CsWinRTWindowsMetadata>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWinRT">
+      <Version>1.4.1</Version>
```
Comment (hez2010) [inline src/BackgroundTasks/BackgroundTasks.csproj:18]: I think we want to use 2.0.0, otherwise the project doesn't build without a net5.0 runtime.

Comment (yaichenbaum) [reply src/BackgroundTasks/BackgroundTasks.csproj:18]: I don't see any reason not to update 👍

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml.cs]: I'm able to run the app in debug mode without any issues but I think it's worth checking.

**Comment by hez2010** [comment]
I'm constantly hitting exception while navigating upwards, should we do a null check before we do navigation?

```
2022-08-26 11:52:14.3983|ERROR|AppUnhandledException|Catastrophic failure (0x8000FFFF (E_UNEXPECTED))
	System.Runtime.InteropServices.COMException (0x8000FFFF): Catastrophic failure (0x8000FFFF (E_UNEXPECTED))
2022-08-26 11:52:14.5087|ERROR|AppUnhandledException|Object reference not set to an instance of an object.
	System.NullReferenceException: Object reference not set to an instance of an object.
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Microsoft.UI.Xaml.Controls.IFrameMethods.Navigate(IObjectReference _obj, Type sourcePageType, Object parameter, NavigationTransitionInfo infoOverride)
   at Microsoft.UI.Xaml.Controls.Frame.Navigate(Type sourcePageType, Object parameter, NavigationTransitionInfo infoOverride)
   at Files.Uwp.Views.ModernShellPage.Up_Click()
   at Files.Uwp.Views.ModernShellPage.ModernShellPage_UpNavRequested(Object sender, EventArgs e)
   at Files.Uwp.ViewModels.ToolbarViewModel.<.ctor>b__180_2(RoutedEventArgs e)
   at CommunityToolkit.Mvvm.Input.RelayCommand`1.Execute(Object parameter)
   at ABI.System.Windows.Input.ICommand.Vftbl.Do_Abi_Execute_3(IntPtr thisPtr, IntPtr parameter)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Windows.ApplicationModel.Core.IUnhandledErrorMethods.Propagate(IObjectReference _obj)
   at Windows.ApplicationModel.Core.UnhandledError.Propagate()
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelperWinUI.<.ctor>b__0_3(Object sender, UnhandledErrorDetectedEventArgs eventArgs)
--- End of stack trace from previous location ---
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelperWinUI.<.ctor>b__0_3(Object sender, UnhandledErrorDetectedEventArgs eventArgs)
   at WinRT.EventSource__EventHandler`1.EventState.<GetEventInvoke>b__1_0(Object obj, T e)
   at ABI.System.EventHandler`1.Do_Abi_Invoke[TAbi](Void* thisPtr, IntPtr sender, TAbi args)
```

**Comment by yaichenbaum** [comment]
Let's note that in the #9780 issue (feel free to add it to the issue description). A null check would be useful but we should probably see why it's null as that shouldn't be the case.

**Comment by gave92** [comment]
> I've bumped the dotnet sdk version to adopt some bug/security fixes.

We're stuck with .100 because on main the fulltrust process doesn't work otherwise.

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs]: Can't repro the crash anymore 👍

File: src/Files.Uwp/Themes/Discord.xaml (hunk 11 above)
Comment (gave92) [reply src/Files.Uwp/Themes/Discord.xaml]: Custom themes restore. Exceptions are the acrylic-glass-mica imitations themes where I had to remove the BackgroundSource property from the Acrylic brushes (not supported)

Comment (yaichenbaum) [reply src/Files.Uwp/Themes/Discord.xaml]: We should definitely remove the acrylic themes.

File: src/Files.Uwp/App.xaml.cs (hunk 3 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs]: Checked again and I can't repro the crash anymore 👍

File: src/Files.Uwp/App.xaml.cs.old (hunk 5 above)
Comment (gave92) [reply src/Files.Uwp/App.xaml.cs.old:1]: Continue where you left off works fine for me with latest commits, could you check?

Comment (yaichenbaum) [reply src/Files.Uwp/App.xaml.cs.old:1]: Works for me as well 👍

**Review by Aei16 — APPROVED** [review]

File: src/Files.Uwp/Views/MainPage.xaml.cs [hunk 22]
```diff
@@ -139,6 +160,48 @@ private void HorizontalMultitaskingControl_Loaded(object sender, RoutedEventArgs
...
+            const uint MDT_Effective_DPI = 0;
+
+            var displayArea = DisplayArea.GetFromWindowId(App.Window.AppWindow.Id, DisplayAreaFallback.Primary);
+            var hMonitor = Win32Interop.GetMonitorFromDisplayId(displayArea.DisplayId);
+            var hr = NativeWinApiHelper.GetDpiForMonitor(hMonitor, MDT_Effective_DPI, out var dpiX, out _);
+            if (hr != 0)
+                return;
+            
+            var scalePercent = (uint)(((long)dpiX * 100 + (96 >> 1)) / 96); // wtf
+            var scaleAdjustment = scalePercent / 100.0;
```
Comment (BreeceW) [inline src/Files.Uwp/Views/MainPage.xaml.cs:193]: You should be able to simplify this by replacing all the DPI code with setting `scaleAdjustment` to `XamlRoot.RasterizationScale`.

**Review by gave92 — APPROVED** [review]

File: src/Files.Uwp/Views/MainPage.xaml.cs (hunk 22 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Views/MainPage.xaml.cs:193]: @d2dyno1

**Review by gave92 — APPROVED** [review]

File: src/Files.Uwp/Views/MainPage.xaml.cs (hunk 22 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Views/MainPage.xaml.cs:193]: This should be addressed with #9945
